analysis.py

```diff
@@ -1,160 +1,536 @@
-    for io_a in io_approach:
-        for _, row in analysis_df.iterrows():
             try:
                 print(f"Calculating indirect impacts for {row['country']} {row['sector']}...")
+                imp = Impact.from_hdf5(row['yearset_path'])
+                if not imp.at_event.any():
+                    # TODO return an object with zero losses so that there's data
+                    print("No non-zero impacts. Skipping")
+                    continue
                 supchain = supply_chain_climada(
                     get_sector_exposure(sector=row['sector'], country=row['country']),
-                    row['impact_yearset'],
+                    imp,
                     impacted_sector=row['sector'],
                     io_approach=io_a
                 )
                 # save direct impacts to a csv
+                # TODO: also save to S3
                 dump_direct_to_csv(
                     supchain=supchain,
-                    haz_type=row['haz_type'],
+                    haz_type=row['hazard'],
                     sector=row['sector'],
-                    scenario=scenario,
-                    ref_year=ref_year,
+                    scenario=row['scenario'],
+                    ref_year=row['ref_year'],
                     country=row['country'],
-                    n_sim=n_sim_years,
+                    n_sim=config['n_sim_years'],
                     return_period=100,
                     output_dir=direct_output_dir
                 )
                 # save indirect impacts to a csv
+                # TODO: also save to S3
                 dump_supchain_to_csv(
                     supchain=supchain,
-                    haz_type=row['haz_type'],
+                    haz_type=row['hazard'],
                     sector=row['sector'],
-                    scenario=scenario,
-                    ref_year=ref_year,
+                    scenario=row['scenario'],
+                    ref_year=row['ref_year'],
                     country=row['country'],
-                    n_sim=n_sim_years,
+                    n_sim=config['n_sim_years'],
                     return_period=100,
                     io_approach=io_a,
                     output_dir=indirect_output_dir
                 )
+                df.loc[i, '_indirect_exists'] = True
 
-            except ValueError as e:
+            except Exception as e:
                 print(f"Error calculating indirect impacts for {row['country']} {row['sector']}:")
                 print("".join(traceback.format_exception(type(e), e, e.__traceback__)))
                 print(e)
 
+    # Run the Supply Chain for each country and sector and output the data needed to csv
+    if DO_INDIRECT:
+        # for io_a in config['io_approach']:
+        for io_a in config['io_approach']:
+            if False or DO_PARALLEL:  # Actually it looks llike a serial calculation is just as efficient
```

Comment:
The `False or` part can go away

Reply:
Raised in https://github.com/ChrisFairless/nccs-supply-chain/issues/41